spark3/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -268,14 +329,19 @@ public void renameTable(Identifier from, Identifier to) throws NoSuchTableExcept
   @Override
   public void invalidateTable(Identifier ident) {
     try {
-      icebergCatalog.loadTable(buildIdentifier(ident)).refresh();
+      TableIdentifier tableIdentifier = buildIdentifier(ident);
+      Table table = isPathIdentifier(ident) ?
+          tables.load(tableIdentifier.name()) :
+          icebergCatalog.loadTable(tableIdentifier);
+      table.refresh();
     } catch (org.apache.iceberg.exceptions.NoSuchTableException ignored) {
       // ignore if the table doesn't exist, it is not cached
     }
   }
 
   @Override
   public Identifier[] listTables(String[] namespace) {
+    // no way to identify if this is a path and we should use tables instead of catalog.
```

Comment:
There should also be no way to pass a path as a namespace.

Reply:
not sure how to identify a path here? Single element w/ '/' in it? Seems arbitrary. The `HadoopCatalog` treats directory names as namespaces so there would be no way to identify that here.

Reply:
Sorry, I wasn't very clear. If anything calls this, then it went through a path for normal identifiers. So I don't think that there is a need to worry about paths here because there isn't a way to pass a path in here.

Reply:
Ah I see, have removed the comment.